### DISCUSSION TITLE
Deep partial

### DISCUSSION THREAD

**Discussion by sbreiler** [open-circle/valibot#963] [Ideas]
I needed a way to make nested objects partial (the ts-typing is off)
```typescript
const valibotDeepPartial = <T extends v.AnySchema>(schema: T): T => {
  // maybe you need to clone schema to not change the original schema?!

  if ('pipe' in schema) {
    schema.pipe = schema.pipe.map(valibotDeepPartial);
  }

  if ('wrapped' in schema) {
    schema.wrapped = valibotDeepPartial(schema.wrapped);
  }

  if (v.isOfType('object', schema) || v.isOfType('strict_object', schema)) {
    for (const [key, propSchema] of Object.entries(schema.entries)) {
      schema.entries[key] = valibotDeepPartial(propSchema);
    }

    schema = v.partial(schema);
  }

  return schema as T;
};
```

[Playground](https://valibot.dev/playground/?code=JYWwDg9gTgLgBAKjgQwM5wG5wGZQiOAcg2QBtgAjCGQgbgCh6BjCAO1XhPKpgBEBTfmAAKyWMDJwAvHAA8AFTj8AHjH6sAJugwA6AIKsAngGUmAC34hkAPgAUqc5eQAuOPICUrxVOtwA3vRwcAD0wXBWhhT8cIYQAK5wrIIacDAQcEykbNEOFlap6azUGWbIrADm0TAWcNDA5cCskrlOAPwAhIxBwNhwtoRgwGD8hHCNcC1W7v6BQROOVjqDw9LzechLQ-w6VmC2XJTUAkKi4mTuDEEAvl1jvf0A7lDIYMMao+OTyNMBc2tOOieLzeqwOPGOIjEMAkpHsCw2QNe-A0F1mN1mPT6umAqAA8th5IZhv0IBQAFb8Jg0AA0-ymcAAPgzMDocfjCcTCBwoMAqQB9UkUqmEWlfdw-WZBbDQPosdjwADaAGt+IZaWA8GBTOsALq1Xq48mUmA6dQwHn8VBw9am1jm4CW8UzP5BL62+2W5WqvUyMFHQSQs6wjUQLXw1F-dF-L6gpZQmHWpwRuBRuBQfgwOJQVh05AodDyBhXBjMNgcODapygsiHPgB07Qsi2Wa6bm8mCGoUwWy-OZxVD8KCuVv2qmd409yVzJzAUjDzbEkc8iq2dy03Qz2Hi6lToJgNCoB7QDTz5b8fY6NsrtcskCNAAy6nK1VsAA5t1OrmupyAMy4WW2Y5GlSk4uhk6bIGoGh6DAp5bBeV7lKu647A+T4vu+36RlhKbfqipbymmlpxKQ8C+peyDYPwpwDrYlZWLSvb9oOri9kEm6uIQZJlPwAACKjIOApDbCwIAilOoRwPuqCHsenEAIwAEwAMwACwAKwAGwAOyvuJ1w7kEkm-jA-5+FcO5fiWcqoBAwk6FkSHpqgJEwKiQA)

Maybe this would be an useful util function? The original [`v.partial`](https://valibot.dev/api/partial/) has additional feature of defining specific key's to be partial, don't know how or even if this is useful for this usecase.

**Comment by fabian-hiller** [comment]
Hey, thank you for reaching out! Your code will most likely produce errors at runtime. A simple example would be a pipeline that accesses certain object properties and expects them to be present, but by making the object partial, they may not be present at runtime, causing errors.  There are probably many more such cases. We might look into implementing a `deepPartial` method next year if more people ask for such a feature.
